src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/MethodToStateMachineRewriter.cs

```diff
@@ -739,8 +753,28 @@ public override BoundNode VisitTryStatement(BoundTryStatement node)
             return result;
         }
 
-        public override BoundNode VisitThisReference(BoundThisReference node)
+        protected BoundStatement CacheThisIfNeeded()
         {
+            // restore "this" cache, if there is a cache
+            if (this.cachedThis != null)
+            {
+                CapturedSymbolReplacement proxy = proxies[this.OriginalMethod.ThisParameter];
+                var fetchThis = proxy.Replacement(F.Syntax, frameType => F.This());
+                return F.Assignment(F.Local(this.cachedThis), fetchThis);
+            }
+
+            // do nothing
+            return F.Block();
```

Comment:
>return F.Block(); [](start = 12, length = 17)

Could we return an empty statement list instead? Might be cheaper in terms of memory.